Source/DafnyRuntime/DafnyRuntimeRust/src/lib.rs

```diff
@@ -34,12 +40,55 @@ pub use num::ToPrimitive;
 pub use num::Zero;
 pub use std::convert::Into;
 
+pub use ::std::any::Any;
+pub use ::std::marker::Send;
+pub use ::std::marker::Sync;
+
+#[cfg(not(feature = "sync"))]
```

Comment:
FYI this would probably be cleaner if you used https://crates.io/crates/cfg-if to group things into just two blocks:

```rust
cfg_if::cfg_if! {
    if #[feature = "sync"] {
        pub type DynAny = dyn Any + Send + Sync;
        pub struct UnsafeCell<T: ?Sized> {
          data: ::std::cell::UnsafeCell<T>, // UnsafeCell for interior mutability
        }
        ...
    } else {
        pub type DynAny = dyn Any;
        pub use ::std::cell::UnsafeCell;
        ...
    }
}
```

Reply:
I would like to avoid taking dependencies that are not strictly necessary. I'm ok with repeating the test in front of every specific declaration as well.